### PR TITLE
allow versioned symlinks

### PR DESCRIPTION
**Pull request by jhheider**
- [x] support semver ranges on `tea -X`
- [x] support semver ranges on symlinks
- [x] support links to semver ranged symlinks (yup, really: `node > node^16 > tea`)
- [x] expand symlink README documentation to cover new feature
- [x] implement test for new feature

### CODE REVIEW

**Comment by jhheider** [comment]
Ugh. Typo in the readme. Need to reverse ln arguments in the second invocation in the script. Tomorrow.

**Comment by mxcl** [comment]
Don’t see the harm in it and probably will be considered a nice feature by many. Though we should document that `python3` == `python^3` (we already support that).

File: README.md [hunk 1]
```diff
@@ -189,6 +189,18 @@ Notably if you create a symlink of `foo` to `tea` (or `tea_foo`) we will
...
 completely transparent to your everyday workflows.
 
+Even better, you can include semver ranges in your link name, so `ln -s tea node^16`
+will let you run `node^16` via the symlink of the same name. For even, even more
+goodness, we will follow a symlink one deep to see if a base link goes to a versioned
+link. Example:
+
+```sh
+$ ln -s tea node^16
+$ ln -s node node^16
```
Comment (mxcl) [inline README.md:199]: aaah nice thinking!

File: tests/integration/tea-XX.test.ts [hunk 2]
```diff
@@ -23,3 +23,10 @@ Deno.test("tea -SX python3.11", async () => {
     assertEquals(out, "3\n")
   })
 })
+
+Deno.test("tea -SX node^16", async () => {
```
Comment (mxcl) [inline tests/integration/tea-XX.test.ts:27]: solid addition

**Comment by jhheider** [comment]
> Though we should document that `python3` == `python^3` (we already support that).

Added that, and cleaned up my typo, as well as using explicit paths to avoid confusion.

**Comment by mxcl** [comment]
Wonderful additions Jacob

**Pull request merged**